### PR TITLE
Odyssey Stats: Add a dismissable nudge for customer feedback after three views of Odyssey Stats

### LINKED ISSUE #72927: Odyssey Stats: Add a dismissable nudge for customer feedback after three views of Odyssey Stats
Image

Link to a Jetpack redirect in the banner with an appropriate slug, like `jetpack-stats-2023-usage-survey` (PCYsg-pY7-p2). Surveys are still being created. See p1HpG7-khi-p2#comment-60393 for final designs.

### PR DESCRIPTION
**Pull request by annacmc**
Related to #72927 

## Proposed Changes

* Adds a dismissible notice box above the Traffic page highlight section. 
* Notice box displays after 3 views of Odyssey Stats
* Notice box nudges user for customer feedback
* Refactored notices to StatsNotices component
* Refactored backend from JITM to Stat's own

Goal

image

## Testing Instructions

- Setup Odyssey with Option 1 in `PejTkB-3E-p2` <- enable New Stats if not yet enabled
- Open `/wp-admin/admin.php?page=stats`
- Ensure you see the opt out notice

image

- Refresh 3 times
- Ensure you see the new user survey notice

image

- Ensure the 'x', 'Take quick survey' and 'Remind me later all work' <- you'll have to remove `stats_options` in `wp_options` to restart the flow to see it or you could hack the code a little for testing
- Refresh
- Ensure these notices don't come up again
		

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/sites/selectors/has-new-stats-feedback-notice.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import getRawSite from 'calypso/state/selectors/get-raw-site';
+
+/**
+ * Returns whether the feebdack request notice should be shown.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param siteId The site ID.
+ * @returns {?boolean}        hasFeedbackNotice
+ */
+export default function hasFeedbackNewStatsNotice( state, siteId ) {
```
Comment (dognose24) [inline client/state/sites/selectors/has-new-stats-feedback-notice.js]: Could we change the function name to the same as the file name `hasNewStatsFeedbackNotice`?

File: packages/components/src/notice-banner/style.scss [hunk 2]
```diff
@@ -105,13 +105,36 @@
...
+	border-left-color: var(--jp-black);
 
 	.notice-banner__icon {
-		fill: var(--black);
+		fill: var(--jp-black);
+	}
+
+	.notice-banner__action-button,
+	.notice-banner__action-link {
+		cursor: grab;
```
Comment (dognose24) [inline packages/components/src/notice-banner/style.scss]: Is this cursor set to `grab` for some reason rather than `pointer`?

**Review by dognose24 — APPROVED** [review]
Works as described! I left some nitty comments that won't be blockers.

**Review by jsnmoon — COMMENTED** [review]
I tried testing this PR on [a version of Jetpack without the notices API](https://github.com/Automattic/jetpack/commit/51f72b4a05d401c46517fc6bee7cac932c6674c8). Clicking "X" on the opt-out notice crashed Odyssey Stats (white screen of death) and logged these errors in the browser console:

image

Did I run into an edge case here? What's our plan for handling Odyssey running on a Jetpack site without the API endpoint (i.e., before https://github.com/Automattic/jetpack/pull/28857)?

File: packages/components/src/notice-banner/style.scss (hunk 2 above)
Comment (jsnmoon) [reply packages/components/src/notice-banner/style.scss]: I agree with Dognose; shouldn't this be `pointer`?

Comment (kangzj) [reply packages/components/src/notice-banner/style.scss]: Thanks for catching it @dognose24 & @jsnmoon, I was testing out something and left this by accident. Fixed in 27c08afba6

File: client/state/sites/selectors/has-new-stats-feedback-notice.js (hunk 1 above)
Comment (kangzj) [reply client/state/sites/selectors/has-new-stats-feedback-notice.js]: Good catch! Fixed in b38f0918c5

**Comment by kangzj** [comment]
> I tried testing this PR on [a version of Jetpack without the notices API](https://github.com/Automattic/jetpack/commit/51f72b4a05d401c46517fc6bee7cac932c6674c8). Clicking "X" on the opt-out notice crashed Odyssey Stats (white screen of death) and logged these errors in the browser console: @jsnmoon 

Thanks for catching this. Changed the return of the component to be not null which fixes it.

**Review by dognose24 — APPROVED** [review]
Those changes look good to me!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7814634

Thank you @annacmc for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7814634).

**Pull request merged**